File: Swift/Two Pointer/13144.swift
let N = Int(readLine()!)!, arr = readLine()!.split(separator: " ").map{ Int($0)! }
var left = 0, right = 0, s = Set<Int>(), ans = 0
while(left <= right && right < N) {
  if !s.contains(arr[right]) {
    s.insert(arr[right])
    right += 1
    continue
  }
  ans += right - left
  s.remove(arr[left])
  left += 1
}
ans += ((s.count)*(s.count+1))/2
print(ans)

File: Python/투포인터/13144.py
import sys; input = sys.stdin.readline

N = int(input())
arr = list(map(int, input().split(" ")))
s = set([])
left, right = 0, 1
s.add(arr[0])
ans = 0
while(left<=right and right < N):
    if arr[right] not in s:
        s.add(arr[right]); right += 1
    else:
        ans += len(s)
        s.remove(arr[left]); left += 1
ans += (len(s)*(len(s)+1))//2
print(ans)
